src/SFA.DAS.EmployerFinance/Models/AccountPayeScheme.cs

```diff
@@ -29,6 +29,7 @@ internal void Delete(DateTime deleted)
             Deleted = deleted;
         }
 
+        //we could make deletion idempotent, so if a soft deleted entity is deleted, it's a no-op. or is it better to explicitly check and throw??
```

Comment:
personally I feel it's better to check and throw, it should be an exceptional circumstance where an entity that has been soft deleted is called to be soft deleted again, either way we should decide and remove this comment from the code.

Reply:
Here's some [background on the (potential) issue](https://docs.particular.net/nservicebus/handlers/accessing-data?version=core_5).

However, I suspect @elliotritchie has already considered this and having non-idempotent message handlers is not an issue for our setup.

We follow this pattern in all our models/handlers, so it doesn't just apply here. I'll remove this comment and add something to the DevNotes file, to double-check that there's no issue.